File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.kt
class Solution {
    fun solution(players: Array<String>, callings: Array<String>): Array<String> {
        val map = mutableMapOf<String, Int>()
        var callingIndex: Int

        for (i in callings.indices) {
            if (map[callings[i]] == null) {
                callingIndex = players.indexOf(callings[i])

                players[callingIndex] = players[callingIndex - 1]
                map[players[callingIndex - 1]] = callingIndex

                players[callingIndex - 1] = callings[i]
                map[callings[i]] = callingIndex - 1
            } else {
                callingIndex = map[callings[i]]!!

                players[callingIndex] = players[callingIndex - 1]
                map[players[callingIndex - 1]] = callingIndex

                players[callingIndex - 1] = callings[i]
                map[callings[i]] = callingIndex - 1
            }
        }
        return players
    }
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.swift
func solution(_ players:[String], _ callings:[String]) -> [String] {
    var answer = players
    var array: [String: Int] = [:]
    
    for (index, player) in players.enumerated() {
        array[player] = index
    }
    
    for calling in callings {
        if let rank = array[calling] {
            answer[rank] = answer[rank-1]
            array[answer[rank]] = rank
            answer[rank-1] = calling
            array[calling] = rank-1
        }
    }
    
    return answer
}